api/types/extension.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"encoding/json"
+
+	"github.com/gravitational/trace"
+)
+
+var certExtensionTypeName = map[CertExtensionType]string{
+	CertExtensionType_SSH: "ssh",
+}
+
+var certExtensionTypeValue = map[string]CertExtensionType{
+	"ssh": CertExtensionType_SSH,
+}
```

Comment:
Does this map is request. It will add additional maintenance cost and probably will quickly be outdated when a new `CertExtensionType` will be added ? Does `CertExtensionType_value` and `CertExtensionType_name` can be reused here ?

Reply:
They can be used, however it makes the configuration that is available to the user quite ugly as it would require the yaml configuration for each be in all caps. They cant go out of date however as an error will be raised if the enum isn't in the maps.